src/search/merge_and_shrink/transition_system.cc

```diff
@@ -28,47 +25,95 @@ ostream &operator<<(ostream &os, const Transition &trans) {
     return os;
 }
 
-// Sorts the given set of transitions and removes duplicates.
-static void normalize_given_transitions(vector<Transition> &transitions) {
-    sort(transitions.begin(), transitions.end());
-    transitions.erase(unique(transitions.begin(), transitions.end()), transitions.end());
+/*
+  Class for representing groups of labels with equivalent transitions in a
+  transition system. See also documentation for TransitionSystem.
+
+  The local label is in a consistent state if label_group and transitions
+  are sorted and unique.
+*/
+LocalLabelInfo::LocalLabelInfo(
+    LabelGroup &&label_group,
+    std::vector<Transition> &&transitions,
+    int cost)
```

Comment:
Instead of computing cost outside, we could pass the Labels instance and compute cost in the constructor (by calling recompute_cost). This would mean, for one case in FTSFactory, to iterate twice over the label group: once here to compute the cost, and once outside to compute the mapping label_to_local_label.